db/dataset.py

```diff
@@ -36,27 +36,65 @@ def create_from_dict(dictionary, author_id):
 
         result = connection.execute("""INSERT INTO dataset (id, name, description, public, author)
                           VALUES (uuid_generate_v4(), %s, %s, %s, %s) RETURNING id""",
-                       (dictionary["name"], dictionary["description"], dictionary["public"], author_id))
+                                    (dictionary["name"], dictionary["description"], dictionary["public"], author_id))
         dataset_id = result.fetchone()[0]
 
         for cls in dictionary["classes"]:
             if "description" not in cls:
                 cls["description"] = None
             result = connection.execute("""INSERT INTO dataset_class (name, description, dataset)
                               VALUES (%s, %s, %s) RETURNING id""",
-                           (cls["name"], cls["description"], dataset_id))
+                                        (cls["name"], cls["description"], dataset_id))
             cls_id = result.fetchone()[0]
 
             # Removing duplicate recordings
             cls["recordings"] = list(set(cls["recordings"]))
 
             for recording_mbid in cls["recordings"]:
                 connection.execute("INSERT INTO dataset_class_member (class, mbid) VALUES (%s, %s)",
-                               (cls_id, recording_mbid))
+                                   (cls_id, recording_mbid))
 
     return dataset_id
 
 
+def update_dataset_meta(dataset_id, meta):
+    """ Update the metadata (name, description, public) for a dataset.
+
+    Args:
+        dataset_id: id of the dataset to update
+        meta: dictionary of metadata to update
+
+    Valid keys for `meta` are: `name`, `description`, `public`.
+    If one of these keys is not present, the corresponding field
+    for the dataset will not be updated. If no keys are present,
+    the dataset will not be updated.
+    """
+
+    params = {}
+    updates = []
+    if "name" in meta:
+        updates.append("name = :name")
+        params["name"] = meta["name"]
+    if "description" in meta:
+        updates.append("description = :description")
+        params["description"] = meta["description"]
+    if "public" in meta:
+        updates.append("public = :public")
+        params["public"] = meta["public"]
```

Comment:
This can be improved a bit by doing something like this:
```python
param = meta.pop("param", None)
if param is not None:
     updates.append("param = :param")
     params["param"] = meta["param"]
```
After that we can check if all keys in `meta` were valid:
```python
if meta:
    raise ValueError('Unexpected meta value: %r' % meta)
```